packages/js/src/classic-editor/replacement-variables.js

```diff
@@ -0,0 +1,339 @@
+	},
+
+	// Deprecrated variables.
+	currentdate: {
+		name: "currentdate",
+		label: __( "Current date", "wordpress-seo" ),
+		getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.currentdate", "" ),
+		isVisible: false,
+	},
+	currentday: {
+		name: "currentday",
+		label: __( "Current day", "wordpress-seo" ),
+		getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.currentday", "" ),
+		isVisible: false,
+	},
+	currentmonth: {
+		name: "currentmonth",
+		label: __( "Current month", "wordpress-seo" ),
+		getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.currentmonth", "" ),
+		isVisible: false,
+	},
+	currentyear: {
+		name: "currentyear",
+		label: __( "Current year", "wordpress-seo" ),
+		getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.currentyear", "" ),
+		isVisible: false,
+	},
+} ) );
+
+/**
+ * Creates the replacement variable configurations list for posts.
+ *
+ * @see [Overview of available variables]{@link https://yoast.com/help/list-available-snippet-variables-yoast-seo/}
+ *
+ * @returns {ReplacementVariableConfiguration[]} The replacement variable configurations for posts.
+ */
+export const createPostReplacementVariableConfigurations = () => [
+	// Basic variables.
+	...values( pick( getConfigurations(), [
+		"date",
+		"title",
+		"sitename",
+		"sitedesc",
+		"excerpt",
+		"excerpt_only",
+		"category",
+		"primary_category",
+		"searchphrase",
+		"sep",
+	] ) ),
+	{
+		name: "parent_title",
+		label: __( "Parent title", "wordpress-seo" ),
+		getReplacement: () => getParentTitle( document.getElementById( "parent_id" ) ),
+	},
+	{
+		name: "tag",
+		label: __( "Tag", "wordpress-seo" ),
+		getReplacement: () => get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.tag", "" ),
+	},
+
+	// Advanced variables.
+	...values( pick( getConfigurations(), [ "id", "name", "page", "focus_keyphrase" ] ) ),
+
+	// Custom fields.
+	...map(
+		get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.custom_fields", {} ),
+		( value, key ) => ( {
+			name: `cf_${ key }`,
+			label: sprintf(
+				// translators: %s expands to the name of the custom field.
+				__( "%s (custom field)", "wordpress-seo" ),
+				key,
+			),
+			getReplacement: () => value,
+		} ),
+	),
+
+	// Custom taxonomies.
+	...flatten( map(
+		get( window, "wpseoScriptData.analysis.plugins.replaceVars.replace_vars.custom_taxonomies", {} ),
+		( { name, description }, key ) => {
+			// translators: %s expands to the key of the custom taxonomy.
+			const label = __( "%s (custom taxonomy)", "wordpress-seo" );
+
+			return [
+				{
+					name: `ct_${ key }`,
+					label: sprintf( label, key ),
+					getReplacement: () => name,
+				},
+				{
+					name: `ct_desc_${ key }`,
+					label: sprintf( label, key ),
+					getReplacement: () => description,
+				},
+			];
+		},
+	) ),
```

Comment:
Maybe even into two separate functions, since it does two things: creating a replace var for the custom taxonomy name and one for the custom taxonomy description.